src/SkyHope.LibraryManager.WebApi/Controllers/BookController.cs

```diff
@@ -1,12 +1,258 @@
     [Route("[controller]/[action]")]
     public class BookController : Controller
     {
+        private const int maxBookCount = 3;
+        private const int dueInDays = 14;
+        private const decimal lateFeePerDay = 0.5m;
+        private readonly BookRepository _bookRepository;
+        private readonly UserRepository _userRepository;
+        private readonly AuthorRepository _authorRepository;
+        private readonly ILogger<BookController> _logger;
+        public BookController(BookRepository bookRepository, UserRepository userRepository, AuthorRepository authorRepository, ILogger<BookController> logger)
+        {
+            _bookRepository = bookRepository;
+            _userRepository = userRepository;
+            _authorRepository = authorRepository;
+            _logger = logger;
+            _logger.LogDebug(1, "NLog injected into BookController");
+        }
 
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<HttpBook>>> GetAllAsync()
+        {
+            var results = new List<HttpBook>();
+            var allBooks = await _bookRepository.GetAllAsync();
+            var availableBooks = allBooks.Where(b => b.IsAvailable && !b.IsDeleted).ToList();
+
+            if (!availableBooks.Any())
+            {
+                return NotFound();
+            }
+
+            foreach (var book in availableBooks)
+            {
+                results.Add(new HttpBook
+                {
+                    Title = book.Title,
+                    BookId = book.BookId,
+                    AuthorId = book.AuthorId,
+                    DewyClass = book.DewyClass,
+                    Isbn = book.Isbn,
+                    Year = book.Year
+                });
+            }
+            _logger.LogInformation($"GetAllAsync in BooksController returned {availableBooks.Count} available books.");
+            return Ok(results);
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<HttpBook>> GetByIdAsync(int id)
+        {
+            var book = await _bookRepository.GetEntityAsync(id);
+            if (book == null)
+            {
+                return NotFound();
+            }
+
+            var result = new HttpBook
+            {
+                Title = book.Title,
+                BookId = book.BookId,
+                AuthorId = book.AuthorId,
+                DewyClass = book.DewyClass,
+                Isbn = book.Isbn,
+                Year = book.Year
+            };
+
+            return Ok(result);
+        }
+
+        [HttpPatch]
+        public async Task<ActionResult> DeleteAsync(HttpBook book)
+        {
+            if (!await _bookRepository.TryDeleteAsync(book.BookId))
+            {
+                return StatusCode(StatusCodes.Status500InternalServerError, "An unexpected error occurred.");
+            }
+            return Ok();
+        }
+
+        [HttpPut]
+        public async Task<ActionResult> CheckOutAsync(HttpBook book, HttpModels.User user)
```

Comment:
A controller method can only take one object as a parameter because it is created from the request body.
You should make the request body as simple as possible - you don't need to take in an entire book model and an entire user model to check out a book - you only need a Book ID and a User ID.

The Book is the resource so that can be part of the request URL, such as:

PUT /books/1/checkout

with the body just being an object with the User ID:

{
  "UserId": 7
}